tests/cpp/test_preseg_passes.cpp

```diff
@@ -982,4 +983,154 @@ TEST_F(PresegTest, TranslateRepeatToExpand5) {
   EXPECT_EQ(heuristic_param->scheduler_type, SchedulerType::PointWise);
 }
 
+TEST_F(PresegTest, FusionRemoveBroadcastSqueeze0) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigConcreteTensor({2, 3, 4, 5});
+  fusion.addInput(tv0);
+  auto tv1 = broadcast(tv0, {true, false, false, false, false});
+  auto tv2 = relu(tv1);
+  auto tv3 = squeeze(tv2, {0});
+  // specify output permutation;
+  std::vector<IterDomain*> tv3_nhwc = {
+      tv3->axis(0), tv3->axis(2), tv3->axis(3), tv3->axis(1)};
+  tv3->setAllocationDomain(tv3_nhwc, true);
+  fusion.addOutput(tv3);
```

Comment:
This is the reason why we care about allocation domain.

i.e. `tv1->relu->tv2->squeeze->tv3`. Here tv3 has an allocation domain that's a permutation.
when we replace it as `tv1->replayed_squeeze->tv4->replayed_relu->tv5`. We need to ensure that `tv5` has the same allocation domain as with `tv3`, otherwise we are going to change the semantics and return an output with the wrong memory format.

Reply:
I'm not saying we should ignore the allocation domain. I just don't see why having an allocation domain can interfere this translation. Why not just keep using `tv3`? Or, it should also be possible to reproduce the same allocation domain with `tv5`.

Reply:
I mistaken what you meant in your earlier question!

> Why not just keep using tv3?

By `keep using tv3`, do you mean that I can have it replayed as `tv1->replayed_squeeze->tv4->replayed_relu->tv3`, I didn't realized that I can just re-use tv3 here, without needing to create a clone of it. Let me try that...

> Or, it should also be possible to reproduce the same allocation domain with tv5.

Yes. I was just trying to keep it simple. If we want to support general transformations, I think I can just do the same replay I did in #3644 https://github.com/NVIDIA/Fuser/pull/3644/files#diff-abe2e10add90523ff6b18e1dc50da46762420e1011078ba47ab52140dc213b6fR80-R85.